### DISCUSSION TITLE
Mute part of a file during encoding

### DISCUSSION THREAD

**Discussion by jleren** [savonet/liquidsoap#3440] [Q&A] [answered]
Hi,

We have a bit of a different use of Liquidsoap. We use it to encode call recordings into a mixed mono file, and a stereo file with a speaker on left and right. This is then encoded into m4a. This happens non realtime, after the recording has been received.

We are now implementing some new functionality, where we want to mute part of the file, during the encoding process.
I have'nt quite been able to figure out how to do it, as it somehow needs to use the absolute time of the file, as the source.

If for example we have a recording, where credit cards are accepted, we need to mute the parts where that is transferred.
The data we would have available, would for example be:

Total length: 49 seconds.
Mute block 1: 8-11 seconds
Mute block 2: 36-40 seconds.

The script would be generated dynamically, which is how we would add the different blocks, so basically I need to figure out what commands to use for creating the mute blocks.

Thanks for a great project, it keeps impressing me as to what it can do.

Kind Regards
Jon

**Comment by smimram** [comment, marked as answer]
You can use `source.time` to get how many time has elapsed since the beginning of a source and `amplify` to mute it when needed. I would thus do something like this, where `s` is your source
```
s = ...

def amp()
  t = source.time(s)
  if (8. <= t and t <= 11.) or (36. <= t and t <= 40.) then 0. else 1. end
end

s = amplify(amp,s)
outpout(s)
```
Here, `amp` returns 0 or 1 depending on the time elapsed in `s` and we amplify `s` accordingly.

**Reply by jleren** [reply]
Thank you for the reply, that was exactly what I was looking for!
